HomeWork6/src/index.ts

```diff
@@ -0,0 +1,118 @@
+interface Movie {
+  title: string;
+  year: number;
+  rating: number;
+  awards: string[];
+}
+interface Category {
+  name: string;
+  movies: Movie[];
+}
+
+// Base class for filters
+class Filter {
+  constructor(
+    public filterType: string,
+    public filter?: string | number,
+    public filterTo?: string | number,
+    public values?: string[]
+  ) {}
+}
+
+// Class for filtering by search
+class SearchFilter extends Filter {
+  constructor(public filter: string) {
+    super('search');
+  }
+}
+
+// Class for filtering by a range of values
+class RangeFilter extends Filter {
+  constructor(
+    public filter: string,
+    public filterTo: number
+  ) {
+    super('range', undefined, filterTo);
+  }
+}
+
+// Class to filter by selected values
+class ValuesFilter extends Filter {
+  constructor(public values: string[]) {
+    super('values', undefined, undefined, values);
```

Comment:
To prevent such construction, you can move the set filters functionality into a method instead of the constructor. This method will allow you to organize the code more effectively and clearly.